doc/source/development/rfc/rfc103_schema_open_option.rst

```diff
@@ -0,0 +1,125 @@
+.. _rfc-103:
+
+===================================================================
+RFC 103 add a OGR_SCHEMA open option to selected OGR drivers
+===================================================================
+
+=============== =============================================
+Author:         Alessandro Pasotti
+Contact:        elpaso at itopen.it
+Status:         Draft
+Created:        2024-10-22
+=============== =============================================
+
+Summary
+-------
+
+This RFC enables users to specify a OGR_SCHEMA open option in the OGR
+drivers that support it.
+
+The new option will be used to override the auto-detected fields types.
+
+Motivation
+----------
+
+Several OGR drivers must guess the attribute data type: CSV, GeoJSON, SQLite,
+the auto-detected types are not always correct and the user has no way to
+override them at opening tim.
+
+For the details please see the discussion attached to the issue: https://github.com/OSGeo/gdal/issues/10943
+
+Implementation
+--------------
+
+A new reserved open option named OGR_SCHEMA will be added to the following drivers:
+
+- CSV
+- GeoJSON
+- SQLite
+- GML
+
+The option will be used to specify the schema in the form of a JSON document (or a path to a JSON file).
+
+The schema document will not need to to contain a definition for every field and it will allow partial overrides of the auto-detected fields types.
+
+The structure of the JSON document will be similar to the one produced by the `ogrinfo -json` command
+with the notable exception that (for the scope of this RFC) only the information related to the type of the fields will be considered:
+
+.. code-block:: json
+
+    {
+    "fields": [
+        {
+        "name": "field1",
+        "type": "string",
+        "subtype": "JSON"
+        },
+        {
+        "name": "field2",
+        "type": "integer",
+        "width":11,
+        "precision":5
+        }
+    ]
+    }
+
+
+In case of multi-layered datasets, the schema will be specified as a list of layers:
+
+.. code-block:: json
+
+    {
+    "layers":[
+        {
+        "name": "layer1",
+        "fields": [
+            {
+            "name": "field1",
+            "type": "string",
+            "subtype": "JSON"
```

Comment:
Actually I see we have spelled it "subType" in ogrinfo -json output:
```
      "fields":[
        {
          "name":"a",
          "type":"String",
          "subType":"JSON",
          "nullable":true,
          "uniqueConstraint":false
        }
```

Schema in https://github.com/OSGeo/gdal/blob/master/apps/data/ogrinfo_output.schema.json#L117

Reply:
I have added a JSON schema to the RFC: https://github.com/OSGeo/gdal/pull/11071/files#diff-9d0c9dc12d0b4a70d63be37cb390f1e9970d226296c30e8883c73a911c82806bR55